lib/connection.ts

```diff
@@ -91,8 +91,8 @@ export class Connection {
     //  - setupRetry: amount of ms to retry when unable to connect on initial setup
     //  - createSocket: create a new Socket connection
     this.options = options;
-    // id if next command to send
-    this.commandId = 1;
+    // id of next command to send
+    this.commandId = 0;
```

Comment:
This is not the right fix, as when we reset it we still set it to 0. Just move the `++` in genCmdId` should fix it

Reply:
If you monitor the WS messages you will see first message will send with commandId: 1, because genCmdId will increase the commandId, and when it set 1 at the beginning the first command will got 2.

Reply:
And what is it set to when the connection adopts a new socket? 

If you move the `++` in genCmdId you return the old value instead of the new one.

Reply:
Yea, Its alternative solution.

Reply:
Your fix is incorrect, when the connection adopts a new socket it is still set to 1.